firmware/powerboard2/code/app/ADC.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * ADC.c
+ */
+
+#include "ADC.h"
+
+/* INCLUDES */
+#include "stm32f4xx.h"
+#include <stdbool.h>
+
+/* DEFINES */
+
+/* TYPEDEFS */
+typedef struct
+{
+    uint16_t rawData[ADC_CHANNEL_COUNT];
+} ADC_data_S;
+
+/* PRIVATE DATA DECLARAION */
+static ADC_data_S ADC_data;
+
+static const uint32_t ADC_softwareChannelToADCChannelMapping[ADC_CHANNEL_COUNT] =
+{
+    [ADC_CHANNEL_ADC_BAT0_1]          = ADC_Channel_0,
+    [ADC_CHANNEL_ADC_BAT0_2]          = ADC_Channel_1,
+    [ADC_CHANNEL_ADC_BAT0_3]          = ADC_Channel_2,
+    [ADC_CHANNEL_ADC_BAT0_4]          = ADC_Channel_3,
+    [ADC_CHANNEL_ADC_BAT0_5]          = ADC_Channel_4,
+    [ADC_CHANNEL_ADC_BAT0_6]          = ADC_Channel_5,
+	[ADC_CHANNEL_ADC_BAT1_1]          = ADC_Channel_6,
+	[ADC_CHANNEL_ADC_BAT1_2]          = ADC_Channel_7,
+    [ADC_CHANNEL_ADC_BAT1_3]          = ADC_Channel_8,
+    [ADC_CHANNEL_ADC_BAT1_4]          = ADC_Channel_9,
+    [ADC_CHANNEL_ADC_BAT1_5]          = ADC_Channel_10,
+	[ADC_CHANNEL_ADC_BAT1_6]          = ADC_Channel_11,
+	[ADC_CHANNEL_ADC_EXT_PRESSURE]    = ADC_Channel_12,
+};
+
+/* PRIVATE FUNCTION DECLARATION */
+static void ADC_private_initGPIOs(void);
+static void ADC_private_initADC(void);
+static void ADC_private_initDMA(void);
+
+/* PRIVATE FUNCTION DEFINITION */
+static void ADC_private_initGPIOs(void)
+{
+    //Enable peripheral clock for GPIOA, GPIOB, and GPIOC
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOA, ENABLE);
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOB, ENABLE);
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOC, ENABLE);
+
+    GPIO_InitTypeDef GPIOA_InitStructure;
+    GPIO_StructInit(&GPIOA_InitStructure);
+
+    GPIOA_InitStructure.GPIO_Pin     = GPIO_Pin_0 | GPIO_Pin_1 | GPIO_Pin_2 | GPIO_Pin_3 | GPIO_Pin_4 | GPIO_Pin_5 | GPIO_Pin_6 | GPIO_Pin_7;
+    GPIOA_InitStructure.GPIO_Mode    = GPIO_Mode_AN;
+    GPIOA_InitStructure.GPIO_OType   = GPIO_OType_PP;
+    GPIOA_InitStructure.GPIO_Speed   = GPIO_Speed_50MHz;
+    GPIOA_InitStructure.GPIO_PuPd    = GPIO_PuPd_NOPULL;
+
+    GPIO_Init(GPIOA, &GPIOA_InitStructure);
+
+    GPIO_InitTypeDef GPIOB_InitStructure;
```

Comment:
You don't have to create a new init variable for port B. After you call `GPIO_Init` for GPIOA, you just have to change what's different on that old struct, then call `GPIO_Init` again for port B.